src/pkcs11/core_pkcs11_test.c

```diff
@@ -0,0 +1,2647 @@
+
+    if( TEST_PROTECT() )
+    {
+        xResult = mbedtls_mpi_read_binary( &xRsaContext.N, xModulus, ulModulusLength );
+        TEST_ASSERT_EQUAL( 0, xResult );
+        xResult = mbedtls_mpi_read_binary( &xRsaContext.E, xExponent, ulExponentLength );
+        TEST_ASSERT_EQUAL( 0, xResult );
+        xRsaContext.len = pkcs11RSA_2048_SIGNATURE_LENGTH;
+        xResult = mbedtls_rsa_check_pubkey( &xRsaContext );
+        TEST_ASSERT_EQUAL( 0, xResult );
+        xResult = mbedtls_rsa_pkcs1_verify( &xRsaContext, NULL, NULL, MBEDTLS_RSA_PUBLIC, MBEDTLS_MD_SHA256, 32, xHashedMessage, xSignature );
+        TEST_ASSERT_EQUAL_MESSAGE( 0, xResult, "mbedTLS failed to parse valid RSA key (verification)" );
+        /* Verify the signature with the generated public key. */
+        xResult = pxGlobalFunctionList->C_VerifyInit( xGlobalSession, &xMechanism, xPublicKeyHandle );
+        TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to VerifyInit RSA." );
+        xResult = pxGlobalFunctionList->C_Verify( xGlobalSession, xPaddedHash, pkcs11RSA_2048_SIGNATURE_LENGTH, xSignature, xSignatureLength );
+        TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to Verify RSA." );
+    }
+
+    mbedtls_rsa_free( &xRsaContext );
+}
+
+/*-----------------------------------------------------------*/
+
+TEST( Full_PKCS11_RSA, AFQP_DestroyObject )
+{
+    CK_RV xResult = CKR_OK;
+
+    xResult = prvDestroyTestCredentials();
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to destroy RSA credentials." );
+}
+
+/*--------------------------------------------------------*/
+/*-------------- EC Tests ------------------------------- */
+/*--------------------------------------------------------*/
+
+TEST_SETUP( Full_PKCS11_EC )
+{
+    CK_RV xResult;
+
+    xResult = xInitializePKCS11();
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to initialize PKCS #11 module." );
+
+    xResult = xInitializePkcs11Session( &xGlobalSession );
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to open PKCS #11 session." );
+}
+
+/*-----------------------------------------------------------*/
+
+TEST_TEAR_DOWN( Full_PKCS11_EC )
+{
+    CK_RV xResult;
+
+    xResult = pxGlobalFunctionList->C_CloseSession( xGlobalSession );
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to close session." );
+
+    xResult = pxGlobalFunctionList->C_Finalize( NULL );
+    TEST_ASSERT_EQUAL_MESSAGE( CKR_OK, xResult, "Failed to finalize session." );
+}
+
+/*-----------------------------------------------------------*/
+
+TEST_GROUP_RUNNER( Full_PKCS11_EC )
+{
+    #if ( PKCS11_TEST_EC_KEY_SUPPORT == 1 )
+        /* Reset the cryptoki to uninitialize state. */
+        prvBeforeRunningTests();
+
+        #if ( PKCS11_TEST_GENERATE_KEYPAIR_SUPPORT == 1 )
+            RUN_TEST_CASE( Full_PKCS11_EC, AFQP_GenerateKeyPair );
+        #endif
+
+        /* Always run AFQP_CreateObject after AFQP_GenerateKeyPair if the port

```

Comment:
Why does the order of the tests here matter ?

Reply:
Some test cases assume the key is imported instead of generated.

Reply:
Resolve the dependency in new commit also remove the comment.